kubernetes.prometheus.autoscale/README.md

```diff
@@ -0,0 +1,285 @@
+zilla   Deployment/zilla   7/10      1         5         3          7m14s
+
+Deployment:
+NAME    READY   UP-TO-DATE   AVAILABLE   AGE
+zilla   3/3     3            3           7m15s
+
+Pods:
+NAME                     READY   STATUS    RESTARTS   AGE
+zilla-6db8d879f5-2wxgw   1/1     Running   0          7m15s
+zilla-6db8d879f5-9bnkh   1/1     Running   0          75s
+zilla-6db8d879f5-fmgqx   1/1     Running   0          75s
+```
+
+Open 21 more connections to zilla.
+
+```bash
+# Open 21 more instances of netcat in the background.
+$ for i in `seq 1 21`; do nc localhost 8080 &; done
+[77] 77777
+[78] 77776
+[79] 77775
+...
+
+# There should be 42 open connections in the background now.
+$ ps auxw | grep "nc localhost 8080" | grep -v grep | wc -l
+      42
+
+# Wait for a few seconds so the metrics get updated. The value of stream_active_received metric should be 42 for one of the pods.
+$ ./check_metric.sh
+The value of stream_active_received metric
+------------------------------------------
+
+Prometheus API:
+{
+...
+        "metric": {
+          "__name__": "stream_active_received",
+        },
+        "value": [
+          "42"
+...
+}
+
+Kubernetes custom metrics API:
+{
+...
+      "metricName": "stream_active_received",
+      "value": "42",
+...
+}
+
+# Wait for a few seconds so the autoscaler can catch up. The zilla deployment should be soon scaled up to 5 pods.
+$ ./check_hpa.sh
+The status of horizontal pod autoscaling
+----------------------------------------
+
+HorizontalPodAutoscaler:
+NAME    REFERENCE          TARGETS    MINPODS   MAXPODS   REPLICAS   AGE
+zilla   Deployment/zilla   8400m/10   1         5         5          12m
+
+Deployment:
+NAME    READY   UP-TO-DATE   AVAILABLE   AGE
+zilla   5/5     5            5           12m
+
+Pods:
+NAME                     READY   STATUS    RESTARTS   AGE
+zilla-6db8d879f5-2wxgw   1/1     Running   0          12m
+zilla-6db8d879f5-9bnkh   1/1     Running   0          6m3s
+zilla-6db8d879f5-fmgqx   1/1     Running   0          6m3s
+zilla-6db8d879f5-g74hl   1/1     Running   0          63s
+zilla-6db8d879f5-q5fmm   1/1     Running   0          63s
+```
+
+Shut down all the connections.
+
+```bash
+# Shut down all running netcat instances.
+$ ps auxw | grep "nc localhost 8080" | grep -v grep | awk '{print $2}' | xargs kill
+[23]  + 55555 terminated  nc localhost 8080
+[22]  + 55554 terminated  nc localhost 8080
+[21]  + 55553 terminated  nc localhost 8080
+...
+
+# There should be no open connections in the background now.
+$ ps auxw | grep "nc localhost 8080" | grep -v grep | wc -l
+       0
+
+# Wait for a few seconds so the metrics get updated. The value of stream_active_received metric should be 0 for all pods.
+$ ./check_metric.sh
+The value of stream_active_received metric
+------------------------------------------
+
+Prometheus API:
+{
+...
+        "metric": {
+          "__name__": "stream_active_received",
+        },
+        "value": [
+          "0"
+...
+}
```

Comment:
When running the example locally, I'm getting an unusual value in the list after scale-in.
```
Prometheus API:
{
  "status": "success",
  "data": {
    "resultType": "vector",
    "result": [
      {
        "metric": {
          "__name__": "stream_active_received",
          "binding": "tcp_server0",
          "endpoint": "prometheus",
          "exported_namespace": "example",
          "instance": "10.1.0.180:9090",
          "job": "zilla",
          "namespace": "zilla-autoscaling",
          "pod": "zilla-6db8d879f5-66fgt",
          "service": "zilla"
        },
        "value": [
          1682709653.16,
          "0"
        ]
      }
    ]
  }
}
```

Reply:
That's just how prometheus returns values like `"value": [ <timestamp>, <value> ]`. I updated the README showing this with comments.